internal/pkg/config/testdata/src/configv2/discussion/source_examples.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package discussion demonstrates some examples of configuration intended by the new config style
+package discussion
+
+import "os"
+
+// All Tokens are sources, even though the underlying type is a basic kind.
+type Token string
+
+// Since configuration does not specify any field as a source, then any TokenBundle is identified as a source.
+// (Implicitly, every field is a source field.)
```

Comment:
I think this makes sense, but it will need to be clearly documented. As will everything else, but I feel like this could be a gotcha. Most source types contain only a few sensitive fields.

It may be helpful to include a flag that prints what the config will match.

Reply:
> It may be helpful to include a flag that prints what the config will match.

This was the intent in a design of subordinate analyzers.  The `sourceType` analyzer emits exactly what source type and fields are matched by config.  The `source` analyzer emits what `*ssa.Value`s are identified as sources, currently based on type and in future possibly based on identified function return values (e.g. `os.Getenv`).

For complete information, we could produce similar analyzers for `sanitizer` and `sink` function declarations and invocations.  I have long held that my ideal world would be one where `levee` performs taint propagation, with all `ssa.Node`s having already been painted with the relevant information.  In that stance, a `--print-identified=source-types` option would just start `levee` analysis by printing the input it has received.